app/assets/javascripts/controllers/ansible_credentials/ansible_credentials_form_controller.js

```diff
@@ -0,0 +1,130 @@
+ManageIQ.angular.app.controller('ansibleCredentialsFormController', ['$http', '$window', 'credentialId', 'miqService', 'API', function($http, $window, credentialId,  miqService, API) {
+  var vm = this;
+
+  var init = function() {
+    vm.credentialModel = {
+      id: null,
+      name: '',
+      type: '',
+    };
+
+    vm.credential_options = {};
+    vm.select_options = [];
+
+    vm.newRecord = credentialId == 'new';
+    vm.afterGet = false;
+    vm.model = 'credentialModel';
+    ManageIQ.angular.scope = vm;
+    vm.saveable = miqService.saveable;
+
+    miqService.sparkleOn();
+
+    // get credential specific options for all supported credential types
+    API.options('/api/authentications')
+      .then(getCredentialOptions)
+      .catch(miqService.handleFailure);
+
+    if (credentialId != 'new') {
+      API.get('/api/authentications/' + credentialId)
+         .then(getCredentialFormData)
+         .catch(miqService.handleFailure);
+    }
+    else {
+      vm.select_options.push({'label':__('<Choose>'), 'value': ''})
+      // FIXME: this should go away once https://github.com/ManageIQ/manageiq/pull/14483 is merged
+      vm.credentialModel.organization = 1; // work-around, organization id needs to be filled in automatically by the backend
+
+      // credential creation requires manager_resource
+      API.get('/api/providers?collection_class=ManageIQ::Providers::EmbeddedAutomationManager')
+         .then(setManagerResource)
+         .catch(miqService.handleFailure);
+
+      vm.afterGet = true;
+      vm.modelCopy = angular.copy( vm.credentialModel );
+      miqService.sparkleOff();
+    }
+  };
+
+  vm.cancelClicked = function(angularForm) {
+    if (credentialId == 'new') {
+      getBack(__("Creation of new Credential was canceled by the user."), true);
+    } else {
+      getBack(sprintf(__("Edit of Credential \"%s\" was canceled by the user."), vm.credentialModel.name), true);
+    }
+  };
+
+  vm.resetClicked = function(angularForm) {
+    vm.credentialModel = angular.copy( vm.modelCopy );
+    angularForm.$setPristine(true);
+    miqService.miqFlash("warn", __("All changes have been reset"));
+  };
+
+  vm.saveClicked = function(angularForm) {
+    API.put('/api/authentications/' + credentialId, vm.credentialModel)
+       .then(getBack(sprintf(__("Modification of Credential \"%s\" has been successfully queued."), vm.credentialModel.name), false))
+       .catch(miqService.handleFailure);
+  };
+
+  vm.addClicked = function(angularForm) {
+    addCredentialKind();
+    API.post('/api/authentications/', vm.credentialModel)
+       .then(getBack(sprintf(__("Add of Credential \"%s\" has been successfully queued."), vm.credentialModel.name)))
+       .catch(miqService.handleFailure);
+  };
+
+  function getCredentialOptions(response) {
+    Object.assign(vm.credential_options, response.data.credential_types.embedded_ansible_credential_types);
+
+    for (opt in vm.credential_options) {
```

Comment:
Needs to be 
`for (var opt in vm.credential_options) `

(define `opt`, that is)